base/server/src/com/netscape/cms/realm/PKIRealm.java

```diff
@@ -132,10 +145,21 @@ public Principal authenticate(final X509Certificate[] certs) {
 
             return getPrincipal(username, authToken);
 
-        } catch (Exception e) {
+        } catch (EMissingCredential | EInvalidCredentials e) { // authentication failure
 
             logger.warn("Unable to authenticate cert chain: " + e.getMessage());
 
+            signedAuditLogger.log(AuthEvent.createFailureEvent(
+                    auditSubjectID,
+                    IAuthSubsystem.CERTUSERDB_AUTHMGR_ID,
+                    attemptedAuditUID));
+
+            throw null;
```

Comment:
(same comment)
I believe as per the attached tomcat documentation, we should be `return` null and not `throw` null

Reply:
Fixing..